src/main/scala/memeid.scala

```diff
@@ -8,13 +8,35 @@ import cats.instances.long._
 import java.lang.Long
 import java.util.{UUID => JUUID}
 
+abstract class Version
```

Comment:
Why using `abstract class` instead of `sealed class`?

Reply:
Wanted to encode an ADT, I think `sealed trait` is more fitting so I'll change it.